### PR TITLE
[SCB-504] upgrade spring boot version

### PR DESCRIPTION
**Pull request by zhengyangyong**
Signed-off-by: zhengyangyong <yangyong.zheng@huawei.com>

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

Spring Boot and Spring Version match

| Spring Boot Version      | Spring Version         |
| :---------- | :---------- |
| 1.5.8.RELEASE | 4.3.12.RELEASE |
| 1.5.9.RELEASE | 4.3.13.RELEASE |
| 1.5.10.RELEASE | 4.3.14.RELEASE |
| 1.5.11.RELEASE | 4.3.15.RELEASE |
| 1.5.12.RELEASE | 4.3.16.RELEASE |

If Consumer use RestTemplate make a invocation and Producer return 590 Cse Internal Server Error like ：
`ensureServerBlowsUp` in https://github.com/apache/incubator-servicecomb-java-chassis/blob/master/integration-tests/springmvc-tests/src/test/java/org/apache/servicecomb/demo/springmvc/tests/SpringMvcIntegrationTestBase.java

* 4.3.12.RELEASE still throw **UnknownHttpStatusCodeException**
* 4.3.13.RELEASE and 4.3.14.RELEASE change to throw **org.springframework.web.client.ResourceAccessException**
* 4.3.15.RELEASE and 4.3.16.RELEASE won't throw any Exception and direct return `{"message":"Cse Internal Server Error"}`

**Spring Cloud version**
>Finchley builds and works with Spring Boot 2.0.x, and is not expected to work with Spring Boot 1.5.x.
The Dalston and Edgware release trains build on Spring Boot 1.5.x, and are not expected to work with Spring Boot 2.0.x.
The Camden release train builds on Spring Boot 1.4.x, but is also tested with 1.5.x.

### CODE REVIEW

**Comment by zhengyangyong** [comment]
@liubao68 Yes ,but the behavior of RestTemplate had changed so may make some trouble when user upgrade to 1.0.0-m2

**Comment by zhengyangyong** [comment]
Discussion Mail :  https://lists.apache.org/thread.html/cfb55b2f1b4bbaf719c73a4f0796a7247ac77183145487229bbbdd36@%3Cdev.servicecomb.apache.org%3E

**Comment by zhengyangyong** [comment]
Upgrade :
spring to 4.3.16.RELEASE
spring boot to 1.5.12.RELEASE
spring cloud to Edgware.SR3

File: integration-tests/spring-zuul-tracing-tests/src/test/java/org/apache/servicecomb/spring/cloud/zuul/tracing/SpringCloudZuulTracingTest.java [hunk 1]
```diff
@@ -82,15 +82,18 @@ public void tracesCallsReceivedFromZuulToCalledService() throws InterruptedExcep
   @Test
   public void tracesFailedCallsReceivedByZuul() throws InterruptedException {
     ResponseEntity<String> responseEntity = testRestTemplate.getForEntity("/dummy/rest/oops", String.class);
-
-    assertThat(responseEntity.getStatusCode(), is(INTERNAL_SERVER_ERROR));
-
+    try {
+      int code = responseEntity.getStatusCode().value();
+      assertThat(code, is(590));
+      expectFailing(IllegalArgumentException.class);
```
Comment (wujimin) [inline integration-tests/spring-zuul-tracing-tests/src/test/java/org/apache/servicecomb/spring/cloud/zuul/tracing/SpringCloudZuulTracingTest.java]: if throw IllegalArgumentException, then failed?

File: integration-tests/spring-zuul-tracing-tests/src/test/java/org/apache/servicecomb/spring/cloud/zuul/tracing/SpringCloudZuulTracingTest.java [hunk 2]
```diff
@@ -82,15 +82,18 @@ public void tracesCallsReceivedFromZuulToCalledService() throws InterruptedExcep
...
+    } catch (Exception ignore) {
```
Comment (wujimin) [inline integration-tests/spring-zuul-tracing-tests/src/test/java/org/apache/servicecomb/spring/cloud/zuul/tracing/SpringCloudZuulTracingTest.java]: what catch and eat Exception?

File: integration-tests/springmvc-tests/src/test/java/org/apache/servicecomb/demo/springmvc/tests/SpringMvcIntegrationTestBase.java [hunk 3]
```diff
@@ -329,14 +325,10 @@ public void blowsUpWhenFileNameDoesNotMatch() throws Exception {
...
 
-    try {
-      restTemplate.postForObject(
-          codeFirstUrl + "uploadWithoutAnnotation",
-          new HttpEntity<>(map, headers),
-          String.class);
-      expectFailing(UnknownHttpStatusCodeException.class);
-    } catch (RestClientException ignored) {
-    }
+    String result = restTemplate
```
Comment (wujimin) [inline integration-tests/springmvc-tests/src/test/java/org/apache/servicecomb/demo/springmvc/tests/SpringMvcIntegrationTestBase.java]: better change to entity?

File: integration-tests/springmvc-tests/src/test/java/org/apache/servicecomb/demo/springmvc/tests/SpringMvcIntegrationTestBase.java [hunk 4]
```diff
@@ -637,14 +628,8 @@ public void ensureServerWorksFine() throws Exception {
...
   public void ensureServerBlowsUp() {
-    try {
-      restTemplate.getForObject(
-          controllerUrl + "sayhi?name=throwexception",
-          String.class);
-      fail("Exception expected, but threw nothing");
-    } catch (UnknownHttpStatusCodeException e) {
-      assertThat(e.getRawStatusCode(), is(590));
-    }
+    String response = restTemplate.getForObject(controllerUrl + "sayhi?name=throwexception", String.class);
```
Comment (wujimin) [inline integration-tests/springmvc-tests/src/test/java/org/apache/servicecomb/demo/springmvc/tests/SpringMvcIntegrationTestBase.java]: postForEntity

File: parent/pom.xml [hunk 5]
```diff
@@ -69,6 +69,7 @@
     <dependency>
       <groupId>org.hamcrest</groupId>
       <artifactId>hamcrest-all</artifactId>
+      <version>1.3</version>
```
Comment (wujimin) [inline parent/pom.xml]: better to declare in dependency?

File: integration-tests/spring-zuul-tracing-tests/src/test/java/org/apache/servicecomb/spring/cloud/zuul/tracing/SpringCloudZuulTracingTest.java (hunk 1 above)
Comment (liubao68) [reply integration-tests/spring-zuul-tracing-tests/src/test/java/org/apache/servicecomb/spring/cloud/zuul/tracing/SpringCloudZuulTracingTest.java]: expectFailing is very hard to understand how it works

File: integration-tests/springmvc-tests/src/test/java/org/apache/servicecomb/demo/springmvc/tests/SpringMvcIntegrationTestBase.java [hunk 6]
```diff
@@ -329,14 +325,10 @@ public void blowsUpWhenFileNameDoesNotMatch() throws Exception {
...
-      restTemplate.postForObject(
-          codeFirstUrl + "uploadWithoutAnnotation",
-          new HttpEntity<>(map, headers),
-          String.class);
-      expectFailing(UnknownHttpStatusCodeException.class);
-    } catch (RestClientException ignored) {
-    }
+    ResponseEntity<String> response = restTemplate
+        .postForEntity(codeFirstUrl + "uploadWithoutAnnotation", new HttpEntity<>(map, headers), String.class);
+    assertThat(response.getStatusCodeValue(), is(response.getStatusCodeValue()));
```
Comment (wujimin) [inline integration-tests/springmvc-tests/src/test/java/org/apache/servicecomb/demo/springmvc/tests/SpringMvcIntegrationTestBase.java]: self equals self?

Comment (zhengyangyong) [reply integration-tests/springmvc-tests/src/test/java/org/apache/servicecomb/demo/springmvc/tests/SpringMvcIntegrationTestBase.java]: sorry,fixed

**Comment by WillemJiang** [comment]
The Company project test failure when switch to the 1.0.0-m2-SNAPSHOT.

**Comment by zhengyangyong** [comment]
Let me have a look

**Pull request merged**